event_types/event_types.py

```diff
@@ -617,6 +617,90 @@ def bin_data_in_energy(dtf, n_bins=20, log_e_reco_bins=None, return_bins=False):
         return dtf_e
 
 
+def bin_data_in_energy_and_offset(
+    dtf, n_bins=20, log_e_reco_bins=None, offset_bins=None, return_bins=False
+):
+    """
+    Bin the data in dtf to n_bins with equal statistics.
+
+    Parameters
+    ----------
+    dtf: pandas DataFrame
+        The DataFrame containing the data.
+        Must contain a 'log_reco_energy' column (used to calculate the bins).
+    n_bins: int, default=20
+        The number of reconstructed energy bins to divide the data in.
+    log_e_reco_bins: array-like, None
+        In case it is not none, it will be used as the energy bins to divide the data sample
+    offset_bins: array-like
+        Camera-offset bins (in degrees) to divide the data sample.
+
+    return_bins: bool
+        If true, the function will return the log_e_reco_bins used to bin the data.
+
+    Returns
+    -------
+    A dictionary of DataFrames (keys=energy ranges, values=separated DataFrames).
+    """
+
+    dtf_e_offset = dict()
+
+    if offset_bins is None:
+        offset_bins = [0, max(dtf["camera_offset"])]  # Fit all elements in one bin by default.
+    elif any(o_bin < 0 for o_bin in offset_bins):
```

Comment:
These tests are good (and necessary). I think they should be in an external function and expanded. For example, I think you need the require that the bins are in increasing order. Also, we should probably set a maximum which makes sense for CTA cameras, e.g., the last bin can be anything above 4 degrees and we put all reconstructed camera offsets in that last bin. Lastly, why don't we provide also the option, like for energy, to simply give the number of bins we want to have?